test/TestCase.cpp

```diff
@@ -139,3 +138,36 @@ EVMVersionRestrictedTestCase::EVMVersionRestrictedTestCase(std::string const& _f
 	if (!comparisonResult)
 		m_shouldRun = false;
 }
+
+void EVMVersionRestrictedTestCase::processBytecodeFormatSetting()
+{
+	std::optional<uint8_t> eofVersion = solidity::test::CommonOptions::get().eofVersion();
+
+	std::string bytecodeFormatString = m_reader.stringSetting("bytecodeFormat", "legacy");
+	if (bytecodeFormatString == "legacy,>=1" || bytecodeFormatString == ">=1,legacy")
+		return;
+
+	if (!m_shouldRun)
+		return;
+
+	// TODO: This is naive implementation because for now we support only one EOF version.
+	if (bytecodeFormatString == "legacy" && eofVersion.has_value())
+		m_shouldRun = false;
+	else if (bytecodeFormatString == ">=1" && !eofVersion.has_value())
+	{
+		// Do not run if non-EOF test run
+		// TODO: Update if EOF moved to Osaka
+		// EOF only available since Prague
+		solAssert(solidity::test::CommonOptions::get().evmVersion() >= langutil::EVMVersion::prague());
```

Comment:
This assert should depend on the value of the EOF version set on the CLI, not the one set in the test. Having a test with `>=1` when EVM version is set to a non-EOF fork on the CLI is normal - the test should be skipped then. The assert is only against the conflicting combination of CLI settings, which you can assume won't happen due to validations.

You should actually see this assert failing when you run tests. Apparently it didn't and I think that's because your only test has `EVMVersion: >=prague` :) You should add another one with `EVMVersion: <prague`.

So:
```suggestion
		solAssert(!eofVersion.has_value() ||solidity::test::CommonOptions::get().evmVersion() >= langutil::EVMVersion::prague());
```
and then it can safely be outside of the `if`. I'd even put it at the beginning of the function.